pkg/eigenState/stakerShares/stakerShares.go

```diff
@@ -610,88 +653,172 @@ func (ss *StakerSharesModel) HandleStateChange(log *storage.TransactionLog) (int
-			Staker:      newState.Staker,
-			Strategy:    newState.Strategy,
-			Shares:      newState.Shares,
-			BlockNumber: blockNumber,
-		}
+				var err error
+				currStakerShares, err = stakerShares.ToStakerSharesDiff()
+				if err != nil {
+					return nil, err
+				}
+			}
 
-		if existingRecord, ok := mappedRecords[slotId]; ok {
-			existingShares, success := numbers.NewBig257().SetString(existingRecord.Shares, 10)
+			// add the shares to the existing shares
+			shareDeltaShares, success := new(big.Int).SetString(shareDelta.Shares, 10)
 			if !success {
-				ss.logger.Sugar().Errorw("Failed to convert existing shares to big.Int",
-					zap.String("shares", existingRecord.Shares),
-					zap.String("staker", existingRecord.Staker),
-					zap.String("strategy", existingRecord.Strategy),
-					zap.Uint64("blockNumber", blockNumber),
+				return nil, xerrors.Errorf("Failed to convert shares to big.Int: %s", shareDelta.Shares)
+			}
+			currStakerShares.Shares = currStakerShares.Shares.Add(currStakerShares.Shares, shareDeltaShares)
+
+			// update the local cache
+			updatedStakerShares[slotID] = currStakerShares
+		case *SlashDiff:
+			slashDiff := diff.Event.(*SlashDiff)
+			query := `
+				with ranked_staker_delegations as (
+					select
+						staker,
+						operator,
+						delegated,
+						ROW_NUMBER() OVER (PARTITION BY staker ORDER BY block_number desc, log_index desc) as rn
+					from staker_delegation_changes
+					where
+						block_number <= @blockNumber and
+						log_index <= @logIndex
+				),
+				delegated_stakers as (
+					select
+						lsd.staker
+					from ranked_staker_delegations as lsd
+					where
+						lsd.operator = @operator
+						and lsd.rn = 1
+				),
+				ranked_staker_shares as (
+					select
+						ds.staker,
+						ss.strategy,
+						COALESCE(ss.shares, 0) as shares,
+						COALESCE(ss.block_number, 0) as block_number, -- Assign a default block number if needed
+						ROW_NUMBER() OVER (
+							PARTITION BY ds.staker, ss.strategy 
+							ORDER BY ss.block_number desc
+						) AS rn
+					from 
+						delegated_stakers as ds
+					left join 
+						staker_shares as ss
+						on ss.staker = ds.staker
+						and ss.strategy = @strategy
+				),
+				latest_staker_shares as (
+					select
+						ls.staker,
+						ls.strategy,
+						ls.shares,
+						ls.block_number
+					from ranked_staker_shares as ls
+					where
+						ls.rn = 1
 				)
-				continue
+				select * from latest_staker_shares
+			`
+			stakerShares := make([]StakerShares, 0)
+
+			// get the staker shares for the stakers who were delegated to the operator
+			// and update the shares with the new max magnitude
+			res := ss.DB.Raw(query,
+				sql.Named("blockNumber", slashDiff.BlockNumber),
+				sql.Named("logIndex", slashDiff.LogIndex),
+				sql.Named("operator", slashDiff.Operator),
+				sql.Named("strategy", slashDiff.Strategy),
+			).Scan(&stakerShares)
+			if res.Error != nil && res.Error != gorm.ErrRecordNotFound {
```

Comment:
Since you're not just finding a single record, I dont think you'll ever reach an ErrRecordNotFound error. You'll just get an empty array back

Reply:
https://github.com/Layr-Labs/sidecar/pull/68/commits/849bd0db77bd553611d6ccdf776de875882dc9c5